tidy3d/components/simulation.py

```diff
@@ -3054,6 +3055,24 @@ def warn_mode_size(monitor: AbstractModeMonitor, msg_header: str, custom_loc: Li
                     custom_loc = ["monitors", mnt_ind]
                     warn_mode_size(monitor=monitor, msg_header=msg_header, custom_loc=custom_loc)
 
+    def _validate_num_cells_in_mode_source(self) -> None:
+        """Warn if mode sources intersect with a very small number of grid cells."""
+        for source in self.sources:
+            if isinstance(source, ModeSource):
+                disc_grid = self.discretize(source)
+                check_dims = [0, 1, 2]
+                check_dims.pop(source.injection_axis)
```

Comment:
we have a utility method `Geometry.pop_axis` that can do this for us. I think it's safer to use in case we ever want to change conventions (although that's unlikely now)

```py
_, check_dims = ModeSource.pop_axis([0,1,2], axis=source.injection_axis)
```

Also probably would make sense to switch the variable names a bit, we tend to use the convention of `axis` being an `int` in `[0,1,2]` and `dims` being a `str` in `xyz`